test/helpers/kubectl.go

```diff
@@ -4490,3 +4490,36 @@ func (kub *Kubectl) NslookupInPod(namespace, pod string, target string) (err err
 func (kub *Kubectl) CiliumOptions() map[string]string {
 	return kub.ciliumOptions
 }
+
+// WaitForCiliumPodToCompleteRegeneration waits for the cilium endpoint regneration to succeed, until timeout occurs.
+// Endpoint.status.log tracks the regeneration status, if log.Code equals OK and state equals ready
+// then it is considered as success.
+func (kub *Kubectl) WaitForCiliumPodToCompleteRegeneration(ciliumPod string, timeout time.Duration) error {
+	body := func() bool {
+		cmd := fmt.Sprintf("cilium endpoint list -o jsonpath='{[?(@.status.identity.id==%d)].status.log}'", ReservedIdentityHost)
+		res := kub.CiliumExecContext(context.Background(), ciliumPod, cmd)
+		if !res.WasSuccessful() {
+			kub.Logger().Errorf("unable to run command '%s' to retrieve status log of  host endpoint from %s: %s",
+				cmd, ciliumPod, res.OutputPrettyPrint())
+			return false
+		}
+
+		var logs []models.EndpointStatusChange
+		if err := res.Unmarshal(&logs); err != nil {
+			kub.Logger().Errorf("unable to unmarshal EndpointStatusChange slice '%#v': %s", logs, err)
```

Comment:
Should be `%w` for errors?

```suggestion
			kub.Logger().Errorf("unable to unmarshal EndpointStatusChange slice '%#v': %w", logs, err)
```